Dockerfile

```diff
@@ -6,21 +6,20 @@ RUN apt-get update && apt-get install -y openssl
 
 RUN npm i -g pnpm
 
-##### DEPS
-FROM base as deps
-
-WORKDIR /app
+##### BUILD
+FROM base as build
 
-ADD . .
+# you can set the value using --build-arg
+ARG env_file=.docker.env
 
-RUN pnpm i
+WORKDIR /app
 
-##### BUILD
-FROM deps as build
+COPY package.json pnpm-lock.yaml ./
 
-WORKDIR /app
+RUN pnpm install --frozen-lockfile
 
-COPY --from=deps /app /app
+COPY . .
+COPY ${env_file} .env
```

Comment:
we don't need this, before `docker build` I will create `.env.production` file and `next build` will automatically use that

Reply:
reverted